pkg/publicshare/manager/json/json.go

```diff
@@ -414,6 +417,9 @@ func (m *manager) UpdatePublicShare(ctx context.Context, u *user.User, req *link
 
 // GetPublicShare gets a public share either by ID or Token.
 func (m *manager) GetPublicShare(ctx context.Context, u *user.User, ref *link.PublicShareReference, sign bool) (*link.PublicShare, error) {
+	m.mutex.Lock()
```

Comment:
Can we change this to use a `sync.RWMutex` instead? We could `RLock` here for improved performance.

Reply:
We can't, because we call `revokeExpiredPublicShare` inside and it lead to the `m.persistence.Write()` call